### PR TITLE
rename functions that run models

### PR DESCRIPTION
**Pull request by iantaylor-NOAA**
This Pull Request resolves issue #723 by renaming the four functions that run the Stock Synthesis executable to make them shorter and easier to remember by avoiding inconsistent capitalization.

@kellijohnson-NOAA since you've already put some thought into this, would you be willing to review the Pull Request?

Note that this PR is from a new "try2" branch in which I put the old and new function names in the same .R file rather than create separate files as in my first attempt discussed in https://github.com/r4ss/r4ss/issues/723#issuecomment-1197482553.

This change breaks (or breaks even more, since #722 already probably broke it) the {nwfscDiags} package, but I think the fixes shouldn't be hard (see https://github.com/nwfsc-assess/nwfscDiag/issues/18).

Once this is merged (tomorrow at the earliest), I will post a message to the VLab forums to spread the word more widely.

### CODE REVIEW

**Review by kellijohnson-NOAA — APPROVED** [review]
Thanks @iantaylor-NOAA for these changes. I made some comments regarding formatting and the use of {lifecycle}, where much of it pertains to all four of the functions even if I only made the comment once. I am uncertain of what the automatic lintr does so some of the formatting may be cleaned up automatically. 

I think one great addition to these changes would be to create a family tag for these functions that manipulate files and run SS3, e.g.,
```
#' @family run functions
```
or
```
#' @family do functions
```
could be options, though I am not married to the actual text 😉

File: NEWS.md [hunk 1]
```diff
@@ -1,15 +1,23 @@
-# r4ss 1.46.0 (26 July 2022)
+# r4ss 1.46.1 (29 July 2022)
 * Major revamp of all functions that run the Stock Synthesis executable
   https://github.com/r4ss/r4ss/pull/722
-  * `run_SS_models()` is now `run()` and has easier defaults and better
-    support for having the Stock Synthesis executable in your path
+  * Functions that run the executable have been renamed:
+    https://github.com/r4ss/r4ss/issues/723:
+    * run_SS_models() -> [run()] (now has easier defaults and better
```
[the remark this answers is not in the document]
Comment (iantaylor-NOAA) [reply NEWS.md:6]: Text changed. And I forgot that NEWS.md doesn't get processed by roxygen2 (of course).

File: NEWS.md [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
-  * Input `show_in_console = FALSE` will pipe output to a text file
-    keeping R console cleaner while models run
-* Deprecates functions that relate to 3.24 models (although
-  `SS_output()` is still compatible back to version 3.24) https://github.com/r4ss/r4ss/pull/718
+  * Input `show_in_console = FALSE` can be used with all these functions
+    and will pipe output to a text file keeping R console cleaner while
+    models run
+* Deprecates functions that relate to 3.24 models, like
+  `SS_readdat_3.24()` (although `SS_output()` is still compatible back
+  to version 3.24) https://github.com/r4ss/r4ss/pull/718
```
[the remark this answers is not in the document]
Comment (iantaylor-NOAA) [reply NEWS.md:20]: The URLs get formatted automatically on github.com:
(from https://github.com/r4ss/r4ss/blob/rename_try2_issue_723/NEWS.md)
image

File: R/jitter.R [hunk 3]
```diff
@@ -50,7 +68,7 @@
 #' profilesummary[["pars"]]
 #' }
 #'
-SS_RunJitter <- function(dir = getwd(),
+jitter <- function(dir = getwd(),
                          mydir = lifecycle::deprecated(),
```
[the remark this answers is not in the document]
Comment (iantaylor-NOAA) [reply R/jitter.R:72]: Yes it will. I just checked.

File: R/jitter.R [hunk 4]
```diff
@@ -65,14 +83,14 @@ SS_RunJitter <- function(dir = getwd(),
   if (lifecycle::is_present(Intern)) {
     lifecycle::deprecate_warn(
       when = "1.45.1",
-      what = "SS_RunJitter(Intern)",
+      what = "jitter(Intern)",
       details = "Please use 'show_in_console' instead"
     )
```
[the remark this answers is not in the document]
Comment (iantaylor-NOAA) [reply R/jitter.R:88]: I see there's a mix of `details` and `with` throughout {r4ss} as a result of these sections getting copy and pasted a lot. I'll change it throughout the package.

Regarding the `when`, these deprecations got added in versions 1.45.1 and 1.46.0 respectively, so doesn't it make sense to just leave those version numbers in place?

File: R/profile.R [hunk 5]
```diff
@@ -1,6 +1,24 @@
...
+#' @description
+#' `r lifecycle::badge("deprecated")`
+#' SS_profile() has been renamed as [profile()]. See
+#' https://github.com/r4ss/r4ss/issues/723 for more details.
+#' 
+#' @author Ian G. Taylor
+#' @export
+#' @seealso [profile()]
+SS_profile <-
+  function() {
```
[the remark this answers is not in the document]
Comment (iantaylor-NOAA) [reply R/profile.R:12]: Yes, not sure why I had it that way. styler::style_file() didn't add it back when I fixed it.

**Comment by iantaylor-NOAA** [comment]
Thanks for the really valuable review @kellijohnson-NOAA. 

I particularly appreciate learning about the roxygen `@family` tag which I applied to the "run functions", "profile functions", "read/write functions", and "tuning functions". 

The "read/write functions" tag replaced this very clever element of a `@seealso` section (which needed to be updated anyway on account of deprecating the read/write functions for older SS3 versions)
```
#' See the following for other \code{SS_read} functions:
#' `r ls("package:r4ss", pattern = "SS_read[a-z]+$")`.\cr
```
The addition of all those `@family` tags probably should have been done in a separate pull request or at least separate commit to keep things cleaner, but it was nice to just get it done.

**Question**: the one unresolved suggestion was regarding bumping the version number in deprecated arguments. Shouldn't those stay at the the first version in which the argument was deprecated?

**Comment by kellijohnson-NOAA** [comment]
My confusion @iantaylor-NOAA with the version and deprecation is that the suggestion is not available in the earlier version. I am not actually sure what is best and more than likely it won't matter what you choose to do here. Maybe @k-doering-NOAA has better guidance than I do.

**Comment by k-doering-NOAA** [comment]
I think leaving it at the first version in which the argument was deprecated makes the most sense

**Pull request merged**